addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4797,6 +4797,8 @@ registry.ImageOptimize = ImageHandlerOption.extend({
             // Image data-mimetype should be changed to SVG since loadImageInfo()
             // will set the original attachment mimetype on it.
             img.dataset.mimetype = 'image/svg+xml';
+            img.dataset.shapeColors = img.dataset.shapeColors.split(';')
+                .map(color => this._getCSSColorValue(color)).join(';');
```

Comment:
Not sure this is perfect: don't you have the risk of ending up saving an image where the URL is still something with `?c1=o-color-3` but where you replaced the data-shape-colors with the actual CSS color of o-color-3?

= Can't you just convert with `_getCSSColorValue` when you **use** that data-shape-colors value?